.buildkite/snyk_report_pipeline.yml

```diff
@@ -1,7 +1,10 @@
 agents:
-  provider: "gcp"
-  machineType: "n1-standard-4"
-  image: family/core-ubuntu-2204
+  image: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-logstash-ci-no-root"
```

Comment:
I took the liberty of kicking off a docker build of the snyk report pipeline using this PR and it failed because it seemed to require docker. See the logs here:
https://buildkite.com/elastic/logstash-snyk-report-ci/builds/181#018c012e-01e1-420c-952c-654d67034e7c/45-227

AFAICS the script seems to require docker so I am afraid you can't run docker commands within another container.
I suggest you switch to a small VM e.g.

```
  agents:
    provider: gcp
    imageProject: elastic-images-prod
    image: family/platform-ingest-logstash-ubuntu-2204
    machineType: "n2-standard-4"
    diskSizeGb: 120
```

Feel free to use this [internal tool](https://github.com/elastic/ingest-dev/blob/main/engineer-productivity/tools/launch_linux_vm.sh) to test the script run (ofc vault creds will fail), but you can also iterate by kicking off builds via https://buildkite.com/elastic/logstash-snyk-report-ci#new and defining `pull/15610/merge` in the branch field.

Reply:
Thank you @dliappis for your feedback!
Ah I see that I didn't know our current Logstash BK agent doesn't support the docker operations.
I have applied the suggested changes (add teams, change the agent, put back the JDK installation) and [kicked off the BK pipeline](https://buildkite.com/elastic/logstash-snyk-report-ci/builds/183#_), seems working as expected 🟢 
Main optimizations with this change are,
- avoid unnecessary clone operation
- better resource utilization by decreasing the agent spec